blockdag/dag.go

```diff
@@ -1933,6 +1933,20 @@ func (dag *BlockDAG) SubnetworkID() *subnetworkid.SubnetworkID {
 	return dag.subnetworkID
 }
 
+// ForEachHash runs the given fn on every hash that's currently known to
+// the DAG.
+//
+// This function is NOT safe for cuncurrent access.
```

Comment:
How do we ensure it is not run concurrently?

I can think of two ways:
1. Create a dedicated mutex.
2. Assign it to one of the existing mutexes (I guess DagMutex is appropriate?).

I'd really prefer option 2.
This just means to add to the comment "This function must be called with the BlablaMutex held for Reads/Writes"

P.S.
But before deleting - see comment I wrote on where you use ForEachHash